### ISSUE TITLE
Warning IDE1003 from AnalyzerDependencyChecker cannot be suppressed

### ISSUE THREAD

**Issue by weltkante** [dotnet/roslyn#8877]
Warning IDE1003 causes false-positives because we are referencing assemblies which are only present at runtime (and we do have fallback code if they ever shouldn't be available for some reason). Since all of our projects need to refer to our analyzer package this creates a lot of noise when having solutions with many projects.

Right clicking the warning it offers me to add it to a supression file but when I click this it does nothing. The suggestion in issue #7049 to add `[assembly: SuppressMessageAttribute ("Category", "ID:FriendlyName")]` attributes works for all warnings I tested except the IDE1003 warning.

There must be something wrong with this warning that neither suppression files nor suppression attributes work. Note that the warning isn't even displayed for selection in the ruleset editor of VS, it only offers warning IDE0001-IDE0005, so I think the [IDE1001+ warnings](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/Diagnostics/Analyzers/IDEDiagnosticIds.cs) could be affected by the problem.

**Comment by sucaritas** [comment]
Sorry to comment on this closed bug, but I have run into this same problem and it seems i cannot suppress this error. this is the syntax i have
```csharp
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage(category: "Roslyn.HostError", checkId: "IDE1003")]
```
this is in a suppression file in the project but the error still comes up. the state that i a in now is that the build succeeds but the errors still show up int the "Error List" tab of visual studio. 
the dependencies are never used at run-time.  and I dont want to include them in the package that delivers the analyzer. can someone please help me get the right syntax for the suppression. or any other workaround to suppress this error msg.

**Comment by andre-f-paggi** [comment]
Still facing this problem

**Comment by mavasani** [comment]
@sucaritas @andre-f-paggi We have changed the suppression design for VS-only `IDEXXXX` diagnostics to account for some performance regressions. See details [here](https://github.com/dotnet/roslyn/pull/23794#issue-158489792). In short, you can now suppress these diagnostics only via explicit suppression in the ruleset file or specifying `/nowarn:<ID>` on the compiler command line. Please let us know if that doesn't work.

**Comment by andre-f-paggi** [comment]
Thanks for the response!

When I search for the code IDE1003, it does not show on the list.

![image](https://user-images.githubusercontent.com/8538938/36493037-5df45cac-170d-11e8-9e38-7df3a599fdaf.png)

**Comment by mavasani** [comment]
Interesting, that should work, I will investigate. Meanwhile, can you manually edit the ruleset to add the entry?

**Comment by andre-f-paggi** [comment]
It worked now! Thank you very much!

**Comment by abezverkov** [comment]
@andre-f-paggi  what did you have to add to the file?

**Comment by andre-f-paggi** [comment]
@abezverkov This is my .ruleset file.

```xml
<?xml version="1.0" encoding="utf-8"?>
<RuleSet Name="New Rule Set" Description=" " ToolsVersion="15.0">
  <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
    <Rule Id="IDE1003" Action="None" />
  </Rules>
</RuleSet>
```
